### PR TITLE
Categories

### PR DESCRIPTION
**Pull request by Rinawaticodes**

### CODE REVIEW

**Review by rocambille — CHANGES_REQUESTED** [review]

**Comment by rocambille** [comment]
it's nice to request a review, but you didn't chance anything after the feedbacks on this PR

File: backend/database.sql [hunk 1]
```diff
@@ -9,10 +9,16 @@ INSERT INTO user (username, email, password) VALUES ('user', 'wcd@gmail.com', 'a
...
+image varchar(100) not null
 )ENGINE=InnoDB DEFAULT CHARSET=latin1;
 
-INSERT INTO category (title) VALUES ("Gestion de l'énergie"),("Gestion de l'eau"),("Gestion des déchets"),("Biodeversité"), ("Communication et Sensibilitaion"),("Responsabilité sociale") ;
+INSERT INTO category (title, image) VALUES ("Gestion de l'énergie", '/src/assets/images_icons/energie.png'),
+("Gestion de l'eau", '/src/assets/images_icons/water.png'),
+("Gestion des déchets", '/src/assets/images_icons/waste.png'),
+("Biodeversité", '/src/assets/images_icons/recycle.png'), 
+("Communication et Sensibilitaion", '/src/assets/images_icons/communication.png'),
+("Responsabilité sociale", '/src/assets/images_icons/human.png') ;
```
[the remark this answers is not in the document]
Comment (Rinawaticodes) [reply backend/database.sql:21]: i'm sorry, i did git pull instead git push.

File: frontend/src/components/Categories.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        setCategories(response.data);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }, []);
+
+  return (
+    <section>
+      <figure>
```
Comment (rocambille) [inline frontend/src/components/Categories.jsx:25]: a figure is a container for an image: remove this one

**Review by rocambille — CHANGES_REQUESTED** [review]

File: frontend/src/components/Categories.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }, []);
+
+  return (
+    <section className="category">
+      <h2>HÉBERGEMENT</h2>
+      <figure className="categories">
```
Comment (rocambille) [inline frontend/src/components/Categories.jsx:26]: remove this line, and only this line

File: frontend/src/components/Categories.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                src={`${
+                  import.meta.env.VITE_BACKEND_URL ?? "http://localhost:5000"
+                }${category.image}`}
+                alt={category.title}
+              />
+              <figcaption>{category.title}</figcaption>
+            </figure>
+          </Link>
+        ))}
+      </figure>
```
Comment (rocambille) [inline frontend/src/components/Categories.jsx:40]: also remove this line, and only this line

File: frontend/src/components/Instruction.jsx [hunk 5]
```diff
@@ -54,27 +53,30 @@ function Instruction() {
         </li>
       </ul>
       <div className="userButton">
-        <button
-          onClick={() => navigate("/category/hebergement")}
+        <Link
+          to="/categories/:categoryId"
```
Comment (rocambille) [inline frontend/src/components/Instruction.jsx:57]: you must provide a value here for :categoryId. from the previous code, I guess you want to write:

    to="categories/hebergement"

File: frontend/src/components/Instruction.jsx [hunk 6]
```diff
@@ -54,27 +53,30 @@ function Instruction() {
...
           className="userBtn"
+          Hebergement
         >
           Hebergement
-        </button>
-        <button
-          onClick={() => navigate("/category/guide")}
+        </Link>
+        <Link
+          to="/categories/Restaurant"
```
Comment (rocambille) [inline frontend/src/components/Instruction.jsx:65]: no upper case in urls

File: frontend/src/components/Instruction.jsx [hunk 7]
```diff
@@ -54,27 +53,30 @@ function Instruction() {
...
+          Hebergement
         >
-          Guide
-        </button>
-        <button
-          onClick={() => navigate("/category/restaurant")}
+          Restaurant
+        </Link>
+        <Link
+          to="/categories/Guide"
```
Comment (rocambille) [inline frontend/src/components/Instruction.jsx:73]: no upper case in urls

**Review by rocambille — CHANGES_REQUESTED** [review]

**Pull request merged**